src/Metal/MTLRenderCommandEncoder.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.Metal {
+	public static class MTLRenderCommandEncoder_Extensions {
+		public unsafe static void SetViewports (this IMTLRenderCommandEncoder This, MTLViewPort [] viewports)
+		{
+			unsafe
+			{
+				fixed (void* handle = viewports)
+					SetViewports (This, (IntPtr) handle, viewports.Length);
```

Comment:
The _normal_ (generated) null check is done later so `viewports.Length` throws a `NullReferenceException`